core/src/main/scala/tapir/Schema.scala

```diff
@@ -17,7 +17,7 @@ object Schema {
   case object SBoolean extends Schema {
     def show: String = "boolean"
   }
-  case class SObject(info: SObjectInfo, fields: Iterable[(String, Schema)], required: Iterable[String]) extends Schema {
+  case class SObject(info: SObjectInfo, fields: Iterable[(String, Schema)], required: Iterable[String]) extends SObjectable {
```

Comment:
maybe rename:
* `SObject` to `SProductObject`
* `SCoproduct` to `SCoproductObject`
* `SObjectable` to `SObject`?

Reply:
Or maybe even simpler, `SRecord`, `SSum` and `SObject` respectively?

Reply:
I think Record/Sum might be easier to understand :)